examples/objectServerExample/src/main/java/io/realm/examples/objectserver/CounterActivity.kt

```diff
@@ -0,0 +1,178 @@
+    private lateinit var counters: RealmResults<CRDTCounter> // Keep strong reference to counter to keep change listeners alive.
+
+    private val loggedInUser: SyncUser?
+        get() {
+            var user: SyncUser? = null
+
+            try {
+                user = SyncUser.current()
+            } catch (ignore: IllegalStateException) {
+            }
+
+            if (user == null) {
+                startActivity(Intent(this, LoginActivity::class.java))
+            }
+
+            return user
+        }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_counter)
+        counterView = binding.textCounter
+        progressBar = binding.progressbar
+        binding.upper.setOnClickListener { adjustCounter(1) }
+        binding.lower.setOnClickListener { adjustCounter(-1) }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        user = loggedInUser
+        user?.let {
+
+            // Create a RealmConfiguration for our user
+            val config = user!!.createConfiguration(BuildConfig.REALM_URL)
+                    .initialData { realm -> realm.createObject(CRDTCounter::class.java, it.identity) }
+                    .build()
+
+            // This will automatically sync all changes in the background for as long as the Realm is open
+            realm = Realm.getInstance(config)
+
+            counterView.text = "-"
+            counters = realm.where<CRDTCounter>().equalTo("name", it.identity).findAllAsync()
+            counters.addChangeListener { counters, changeSet ->
+                if (counters.isValid && !counters.isEmpty()) {
+                    val counter = counters.first()
+                    counterView.text = String.format(Locale.US, "%d", counter!!.count)
+                } else {
+                    counterView.text = "-"
+                }
+            }
+
+            // Setup progress listeners for indeterminate progress bars
+            session = realm.syncSession
+            session.run {
+                addDownloadProgressListener(ProgressMode.INDEFINITELY, downloadListener)
+                addUploadProgressListener(ProgressMode.INDEFINITELY, uploadListener)
+            }
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        user?.let {
+            session.run {
+                removeProgressListener(downloadListener)
+                removeProgressListener(uploadListener)
+            }
+            realm.close()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_counter, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_logout -> {
+                realm.close()
+                user?.let  {
```

Comment:
Prefer `val user = user; if(user != null) {`

Reply:
I'm not 100% sure why you feel this way, because it feels more verbose? Is it because `let` actually returns a value or...?

Reply:
The idea is that you can avoid `?.let { it ->` with it. But `it` can be named too if you want. It's just that if you name it the same thing, then the lint starts yelling at you that it's "name-shadowing"